src/handlers/mod.rs

```diff
@@ -116,7 +116,13 @@ impl SeatHandler for State {
         let dh = &self.niri.display_handle;
         let client = focused.and_then(|s| dh.get_client(s.id()).ok());
         set_data_device_focus(dh, seat, client.clone());
-        set_primary_focus(dh, seat, client);
+        let client_state = client.as_ref().and_then(|c| c.get_data::<ClientState>());
+        if !client_state
+            .map(|c| c.primary_selection_disabled)
+            .unwrap_or(false)
```

Comment:
Is this check necessary? When it's disabled, the client won't have a global anyway so this should be inert?